processor/src/main/java/org/mapstruct/ap/internal/model/common/Type.java

```diff
@@ -836,4 +848,35 @@ public boolean hasEmptyAccessibleContructor() {
 
         return null;
     }
+
+    public Type getElementType() {
+        if ( isArrayType() ) {
+            return getComponentType();
+        }
+        else {
+            return getTypeBound();
+        }
+    }
+
+    /**
+     * Searches for the given superclass and collects all type arguments for the given className.
+     * This is used for support with Java 8 streams, as must use the FQN in order to be able to compile
+     * with Java 6.
+     *
+     * @param superClassName the superclass or interface the generic type arguments are searched for
+     * @return a list of type arguments or null, if superclass was not found
+     */
+
+    public Collection<Type> determineTypeArguments(String superClassName) {
+        Class<?> superClass = null;
+        try {
+            superClass = Class.forName( superClassName );
```

Comment:
I needed to get the type element of the `java.util.stream.Stream` being mapped. I can try the `Elements#getTypeElement()` to see if it works like that. I didn't know about that. I can't call our `determineTypeArguments(Class<?> superClass)` as the `Stream` does not compile with Java 6 and 7